Sources/SnapshotTesting/Snapshotting/UIImage.swift

```diff
@@ -99,12 +103,51 @@ private func compare(_ old: UIImage, _ new: UIImage, precision: Float) -> Bool {
   guard let newerContext = context(for: newerCgImage, data: &newerBytes) else { return false }
   guard let newerData = newerContext.data else { return false }
   if memcmp(oldData, newerData, byteCount) == 0 { return true }
-  if precision >= 1 { return false }
-  var differentPixelCount = 0
-  let threshold = 1 - precision
-  for byte in 0..<byteCount {
-    if oldBytes[byte] != newerBytes[byte] { differentPixelCount += 1 }
-    if Float(differentPixelCount) / Float(byteCount) > threshold { return false}
+  if precision >= 1, perceptualPrecision >= 1 { return false }
+  if perceptualPrecision < 1, #available(iOS 11.0, tvOS 11.0, *) {
+    let deltaFilter = CIFilter(
+      name: "CILabDeltaE",
+      parameters: [
+        kCIInputImageKey: CIImage(cgImage: newCgImage),
+        "inputImage2": CIImage(cgImage: oldCgImage)
+      ]
+    )
+    guard let deltaOutputImage = deltaFilter?.outputImage else { return false }
+    let extent = CGRect(x: 0, y: 0, width: oldCgImage.width, height: oldCgImage.height)
+    let thresholdOutputImage = try? ThresholdImageProcessorKernel.apply(
+      withExtent: extent,
+      inputs: [deltaOutputImage],
+      arguments: [ThresholdImageProcessorKernel.inputThresholdKey: (1 - perceptualPrecision) * 100]
+    )
+    guard let thresholdOutputImage = thresholdOutputImage else { return false }
```

Comment:
```suggestion
    guard
      let thresholdOutputImage = try? ThresholdImageProcessorKernel.apply(
        withExtent: extent,
        inputs: [deltaOutputImage],
        arguments: [ThresholdImageProcessorKernel.inputThresholdKey: (1 - perceptualPrecision) * 100]
      )
    else { return false }
```

Reply:
Thanks for adding support for older Swift versions!